shotover-proxy/src/transforms/throttling.rs

```diff
@@ -43,10 +43,17 @@ impl Transform for RequestThrottling {
         let throttled_messages: Vec<(Message, usize)> = (0..message_wrapper.messages.len())
             .into_iter()
             .rev()
-            .filter(|_| self.limiter.check().is_err())
-            .map(|i| {
-                let message = message_wrapper.messages.remove(i);
-                (message, i)
+            .filter_map(|i| {
+                if self
+                    .limiter
+                    .check_n(message_wrapper.messages[i].message_count().ok()?)
```

Comment:
Should this be configurable? Or does it make sense to just always enable this logic?
I'm leaning towards just always enable it, but it something we should think about.

Reply:
I think enable it too, the transform is specifically "per query throttling".